src/component/base/__tests__/DraftEditor.react-test.js

```diff
@@ -50,3 +51,62 @@ test('must has editorKey same as props', () => {
     shallow._instance.getEditorKey || shallow._instance._instance.getEditorKey;
   expect(getEditorKey()).toMatchSnapshot();
 });
+
+describe('focus', function() {
+  let props;
+  beforeEach(function() {
+    props = {editorState: EditorState.createEmpty(), onChange: jest.fn()};
+  });
+
+  describe('when already focused', function() {
+    beforeEach(function() {
+      const selection = props.editorState.getSelection();
+      props.editorState = EditorState.forceSelection(
+        props.editorState,
+        selection.merge({hasFocus: true}),
+      );
+    });
+
+    test('must not call update', function() {
+      const selection = props.editorState.getSelection();
+      props.editorState = EditorState.forceSelection(
+        props.editorState,
+        selection.merge({hasFocus: true}),
+      );
+      const instance = renderEditorInContainer(props);
+      instance.focus();
+      expect(props.onChange.mock.calls.length).toMatchSnapshot();
+    });
+
+    test('with newEditorState must call update with newEditorState', function() {
+      const selection = props.editorState.getSelection();
+      const testText = 'example content text';
+      const content = ContentState.createFromText(testText);
+      const newEditorState = EditorState.forceSelection(
+        EditorState.createWithContent(content),
+        selection.merge({hasFocus: true}),
+      );
+      const instance = renderEditorInContainer(props);
+      instance.focus(null, newEditorState);
+      expect(props.onChange.mock.calls.length).toMatchSnapshot();
+      const updatedEditorState = props.onChange.mock.calls[0][0];
+      expect(
+        updatedEditorState.getCurrentContent().getPlainText(),
+      ).toMatchSnapshot();
+    });
+  });
+
+  test('must call update', function() {
+    const instance = renderEditorInContainer(props);
+    instance.focus();
+    expect(props.onChange.mock.calls.length).toMatchSnapshot();
+  });
+});
+
+function renderEditorInContainer(props) {
```

Comment:
nit, would be great if we could declare helper functions as arrow functions and move them to the top of the file.